src/components/Tool/Appeal/List/ContactRow/ContactRow.tsx

```diff
@@ -0,0 +1,130 @@
+import { getLocalizedPledgeFrequency } from 'src/utils/functions/getLocalizedPledgeFrequency';
+import {
+  AppealsContext,
+  AppealsType,
+} from '../../AppealsContext/AppealsContext';
+
+// When making changes in this file, also check to see if you don't need to make changes to the below file
+// src/components/Contacts/ContactRow/ContactRow.tsx
+
+type ContactRow = Pick<
+  Contact,
+  | 'id'
+  | 'name'
+  | 'pledgeAmount'
+  | 'pledgeFrequency'
+  | 'pledgeCurrency'
+  | 'pledgeReceived'
+>;
+interface Props {
+  contact: ContactRow;
+  useTopMargin?: boolean;
+}
+
+export const ContactRow: React.FC<Props> = ({ contact, useTopMargin }) => {
+  const {
+    isRowChecked: isChecked,
+    contactDetailsOpen,
+    setContactFocus: onContactSelected,
+    toggleSelectionById: onContactCheckToggle,
+  } = React.useContext(AppealsContext) as AppealsType;
+  const { t } = useTranslation();
+  const locale = useLocale();
+
+  const handleContactClick = () => {
+    onContactSelected(contact.id);
+  };
+
+  const {
+    id: contactId,
+    name,
+    pledgeAmount,
+    pledgeCurrency,
+    pledgeFrequency,
+  } = contact;
+
+  const pledge = useMemo(
+    () =>
+      pledgeAmount && pledgeCurrency
+        ? currencyFormat(pledgeAmount, pledgeCurrency, locale)
+        : pledgeAmount || currencyFormat(0, pledgeCurrency, locale),
+    [pledgeAmount, pledgeAmount, pledgeCurrency, locale],
+  );
+  const frequency = useMemo(
+    () =>
+      (pledgeFrequency && getLocalizedPledgeFrequency(t, pledgeFrequency)) ||
+      '',
+    [pledgeFrequency],
+  );
+
+  return (
+    <ListItemButton
+      focusRipple
+      onClick={handleContactClick}
+      onMouseEnter={preloadContactsRightPanel}
+      useTopMargin={useTopMargin}
+      isChecked={isChecked}
+      contactId={contactId}
+      data-testid="rowButton"
+    >
+      <Hidden xsDown>
+        <ListItemIcon>
+          <StyledCheckbox
+            checked={isChecked(contact.id)}
+            color="secondary"
+            onClick={(event) => event.stopPropagation()}
+            onChange={() => onContactCheckToggle(contact.id)}
+            value={isChecked}
+          />
+        </ListItemIcon>
+      </Hidden>
+      <Grid container alignItems="center">
+        <Grid item xs={10} md={6} style={{ paddingRight: 16 }}>
+          <ListItemText
+            primary={
+              <Typography component="span" variant="h6" noWrap>
+                <Box component="span" display="flex" alignItems="center">
+                  {name}
+                </Box>
+              </Typography>
+            }
+          />
+        </Grid>
+        <Grid item xs={2} md={6}>
+          <Box
+            display="flex"
+            alignItems="center"
+            justifyContent={contactDetailsOpen ? 'flex-end' : undefined}
+          >
+            <Box display="flex" flexDirection="column" justifyContent="center">
+              <Typography component="span">
+                {`${pledge} ${frequency}`}
```

Comment:
It seems like sometimes it is not showing the actual commitment like it did in Angular. I'm looking at John Doe Lois Lane's account and the End of Year Gift appeal.
<img width="1340" alt="Screenshot 2024-08-02 at 11 49 51 AM" src="https://github.com/user-attachments/assets/74650ce6-84aa-4f36-9aa9-d2983079a70e">

Reply:
I am asking Andrew about this as I'm not able to grab the given total without doing a separate request for each

Reply:
I have fixed this in a later PR